numpy/lib/function_base.py

```diff
@@ -1076,6 +1076,12 @@ def gradient(f, *varargs, axis=None, edge_order=1):
     array([[ 2.,  2., -1.],
            [ 2.,  2., -1.]])
 
+    >>> import numpy as np
+    >>> x = np.array([0, 1, 2, 3, 4])
+    >>> y = x**2
+    >>> np.gradient(y, x, edge_order=2)
+    array([0., 2., 4., 6., 8.])
+
```

Comment:
As currently written, it sounds like your example is for the line " The `axis` keyword can be used to specify a subset of axes of which the gradient is calculated", which isn't the case. It would be helpful if you add a line explaining what your example is trying to show, and ideally add a second example showing what happens if you use `dx = np.gradient(x)` instead of passing `x`.